### ISSUE TITLE
Fail2ban .local file not updated

### ISSUE THREAD

**Issue by Vagrantin** [webmin/webmin#787]
Hello,

I have updated Webmin to version 1.872 and the .local files edition doesn't work properly here is how it behaves:
Debian9  
Apache 2.4.25  
Virtualmin 6.01.gpl  
Webmin 1.872.0121.1610  
Fail2ban 0.9.6  

So for this testing I have created a new ``log filter`` from the interface then created a new ``Filter action jail``  
Regarding the ``Filter action jail`` I don't have any issue the ``myjail.local`` files which was created manually before is properly updated.  

The issue concern the ``log filter``, when you created a new one, two config files are created in ``/etc/fail2ban/filter.d/``, you'll have a ``mylogfilter.conf`` and a ``mylogfilter.local``.  
The ``mylogfilter.conf`` contains only this string ``[Definition]`` and the ``mylogfilter.local`` is completely **empty**.  

One of the consequence is in the interface you insert your regex hit *save* but the regex is never saved, and you find the regex box empty, if you go back in your newly created filter.  
The other issue is if you don't pay attention the rule will never work since the filter is **empty**.  
Also once I have updated manually the ``mylogfilter.local`` file, the UI gets updated, if I do the same for the ``mylogfilter.conf`` file only, the UI doesn't get updated, but this is a normal behavior since ``mylogfilter.local`` supersede ``mylogfilter.conf`` in fail2ban model.  

Hope this is clear enough

EDIT: it seems it's not even possible to update the ``conf`` one, once I got back it from the UI, the config was gone..  So only update the ``.local`` version is working for me.

Matth

**Comment by gnadelwartz** [comment]
jamie will solved the reading/writing of local files in next release.
in current develop relese it should work for jails.

whats missing is applying same logic to filters and actions config files

**Comment by gnadelwartz** [comment]
see also  #764 and #599

**Comment by Vagrantin** [comment]
Ok, all clear, then I'm closing this issue.
Thanks

**Comment by gnadelwartz** [comment]
please reopen until problem is solved in your installation or jamie explicitly close it with an associated patch

**Comment by Vagrantin** [comment]
Ok, understand, as soon as the patch version goes out I'll give a test.

**Comment by gnadelwartz** [comment]
i have parsed the source of fail2ban-lib again. it looks like jamie already fixed the filter and action parsing too.

pls run update-from-repo.sh in your webmin installation dir and check if the issue exist and report if it works now

**Comment by jcameron** [comment]
Yeah, the 1.872 devel release  doesn't include the latest fixes. I will create a 1.873 version that does.

**Comment by Vagrantin** [comment]
Hello, 

Just gave it a try but the update is failing because the install script can't find the ``maketemp.pl`` after it has been untar in the .~files/webadmin/tarballs/webmin-1.873-0128.1555/

And effectively I double checked this perl script is not available in this folder.
So the update didn't goes trough.

Matth

**Comment by gnadelwartz** [comment]
yes there was a broken versiin of the script. copy the script from ,.~files/webadm to . and try again

**Comment by gnadelwartz** [comment]
any progress?

**Comment by Vagrantin** [comment]
I'll test that and report back asap.

**Comment by gnadelwartz** [comment]
from what version you try to update? in dev version 1.872+3 the script was broken.
to download the latest version: `wget https://rawgit.com/webmin/webmin/master/update-from-repo.sh`

**Comment by Vagrantin** [comment]
I'm running webmin 1.87302170857.

There is improvement, at the log filter creation, the ``.local`` version is properly created and updated, but still a ``.conf`` version of the same filter is created but empty, I don't think this is necessary to have the ``.conf`` version only one version is enough or we'll end up we tones of files in this filter folder.

Also when you delete the filter created, only the ``.local`` one is deleted, the ``.conf`` one stays in the filter folder.

hope that help

Matth

**Comment by gnadelwartz** [comment]
Thanks for reporting!

The current rule is: don't touch the .conf files, so its OK that it stay's after removing the .local file.
What ca be questioned is the inital creation of an empty .conf file, but this doesn't harm functionality.

So your initial issue is solved, correct?

**Comment by gnadelwartz** [comment]
I had a look at fail.ban documention again, it's only stated that existing files are read in a specific, alphabetical order, but not that a .conf file must exist to read the corresponding .local file.

https://www.systutorials.com/docs/linux/man/5-jail.conf/
 
IMHO in *.d/ dirs it's not needed to create an empty e.g. myfilter.conf when creating myfilter.local.

**Comment by Vagrantin** [comment]
@gnadelwartz yes I agree this .conf file is not necessary since it's most-likely never been used.
Also I confirm the initial issue is fixed.

**Comment by iliaross** [comment]
❗️ If the issue isn't resolved, feel free to reopen it with updated details.
